modules/core/nuxeo-core-bulk/src/test/java/org/nuxeo/ecm/core/bulk/action/TestWordCountAction.java

```diff
@@ -70,9 +71,57 @@ public void testWordCount() throws Exception {
         assertTrue(bulkService.await(commandId, Duration.ofSeconds(20)));
         BulkStatus status = bulkService.getStatus(commandId);
         assertEquals(COMPLETED, status.getState());
+        assertFalse(status.isLimitedQuery());
         assertEquals(wordCount, status.getResult().get("wordCount"));
     }
 
+    @Test
+    public void testWordCountWithLimitedQery() throws Exception {
+        int wordCount = 2732;
+        String myFile = createFile(wordCount);
+        // Use an action limited by default to the first 100 first lines
+        BulkCommand command = new BulkCommand.Builder("testWordCountLimited", myFile, "system")
+                .useGenericScroller()
+                .build();
+        String commandId = bulkService.submit(command);
+        assertTrue(bulkService.await(commandId, Duration.ofSeconds(20)));
+        BulkStatus status = bulkService.getStatus(commandId);
+        assertEquals(COMPLETED, status.getState());
+        assertTrue(status.isLimitedQuery());
+        int result100 = (Integer) status.getResult().get("wordCount");
+        assertTrue(result100 > 0);
+        // the number of word for the first 100 lines is inferior to the total number of words in the file
+        assertTrue(wordCount > result100);
+
+        // Now set an explicit limit to a lower number of lines
+        command = new BulkCommand.Builder("testWordCountLimited",  myFile, "system")
+                .useGenericScroller()
+                .queryLimit(10)
+                .build();
+        commandId = bulkService.submit(command);
+        assertTrue(bulkService.await(commandId, Duration.ofSeconds(20)));
+        status = bulkService.getStatus(commandId);
+        assertTrue(status.isLimitedQuery());
+        assertEquals(COMPLETED, status.getState());
+        int result10 = (Integer) status.getResult().get("wordCount");
+        assertTrue(result10 > 0);
+        // there is less words in 10 lines than 100
+        assertTrue(result100 > result10);
+
+        // Now request explicit unlimited number of lines
+        command = new BulkCommand.Builder("testWordCountLimited",  myFile, "system")
+                .useGenericScroller()
+                .queryUnlimited()
+                .build();
+        commandId = bulkService.submit(command);
+        assertTrue(bulkService.await(commandId, Duration.ofSeconds(20)));
+        status = bulkService.getStatus(commandId);
+        assertEquals(COMPLETED, status.getState());
+        assertFalse(status.isLimitedQuery());
+        long resultUnlimited = (Integer) status.getResult().get("wordCount");
+        assertEquals(wordCount, resultUnlimited);
+    }
```

Comment:
Please add tests for the behavior where the `.queryLimit()` is wordCount-1, exactly wordCount, and wordCount+1 to test edge cases.